### PR TITLE
Added basic sample app which depends only on storio-sqlite and storio-annotations

### PR DESCRIPTION
**Pull request by skrzyneckik**
At it was requested in #622, I created basic sample based on storio-sample-app. It's lack of documentation as a readme file and comments in code, which can be useful for new library user. Is it good idea to continue working on it?

### CODE REVIEW

**Comment by nikitin-da** [comment]
Thanks a lot, @skrzyneckik!
Can you please squash commits into one?

**Comment by skrzyneckik** [comment]
@nikitin-da Sure, but isn't it required - I've seen that [github added feature to squash commits into one when PR is accepted](https://github.com/blog/2141-squash-your-commits). Let me know if I still need to do it

**Comment by skrzyneckik** [comment]
@nikitin-da Created PR #688 with squashed commits from this PR

**Comment by nikitin-da** [comment]
Oh yeah, I forgot about this feature =(
Ok, but a few nits

**Comment by skrzyneckik** [comment]
@nikitin-da should I close #688 then? Building process of #688 failed because one of storio-sqlite tested has failed and I didn't touch it at all.

**Comment by nikitin-da** [comment]
Yes, there is fluky test on travis =(
Sure we can merge with squash #686 to keep comments. #688 just can be closed (sorry).
@rsinukov PTAL, please ;)

**Comment by skrzyneckik** [comment]
@nikitin-da great! I've closed #688 and #689 - sorry for a mess

**Comment by rsinukov** [comment]
LGTM

**Comment by nikitin-da** [comment]
@rsinukov thanks!
@skrzyneckik can you please fix comments and than we will merge it?

File: storio-basic-sample-app/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/.gitignore]: I believe it's handled by `build/` in root `.gitignore`

File: storio-basic-sample-app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+        testInstrumentationRunner 'android.support.test.runner.AndroidJUnitRunner'
+    }
+
+    lintOptions {
+        disable 'GoogleAppIndexingWarning'
+    }
+}
+
+dependencies {
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/build.gradle]: Please add comment so one could easily understand where to find `libraries` declaration, something like: `// libraries declared in root build.gradle.`

File: storio-basic-sample-app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        <activity android:name="com.pushtorefresh.storio.basic_sample.MainActivity">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN"/>
+
+                <category android:name="android.intent.category.LAUNCHER"/>
+            </intent-filter>
+        </activity>
+    </application>
+
+</manifest>
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/AndroidManifest.xml]: nit: empty line will remove warning from github

File: storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/DbOpenHelper.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.pushtorefresh.storio.basic_sample;
+
+import android.content.Context;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.support.annotation.NonNull;
+
+public class DbOpenHelper extends SQLiteOpenHelper {
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/DbOpenHelper.java]: btw, you could use Kotlin if you want :)

File: storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+public class MainActivity extends AppCompatActivity {
+
+    private StorIOSQLite storIOSQLite;
+
+    private RecyclerView recyclerView;
+    private TweetsAdapter tweetsAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/MainActivity.java]: nullability

File: storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        storIOSQLite = DefaultStorIOSQLite.builder()
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/MainActivity.java]: Lol, already wrote a comment about DI here and only then understood that it's basic sample :)

File: storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    void addTweets() {
+        final List<Tweet> tweets = new ArrayList<Tweet>();
+
+        tweets.add(Tweet.newTweet("artem_zin", "Checkout StorIO — modern API for SQLiteDatabase & ContentResolver"));
+        tweets.add(Tweet.newTweet("HackerNews", "It's revolution! Dolphins can write news on HackerNews with our new app!"));
+        tweets.add(Tweet.newTweet("AndroidDevReddit", "Awesome library — StorIO"));
+        tweets.add(Tweet.newTweet("Facebook", "Facebook community in Twitter is more popular than Facebook community in Facebook and Instagram!"));
+        tweets.add(Tweet.newTweet("Google", "Android be together not the same: AOSP, AOSP + Google Apps, Samsung Android"));
+        tweets.add(Tweet.newTweet("Reddit", "Now we can send funny gifs directly into your brain via Oculus Rift app!"));
+        tweets.add(Tweet.newTweet("ElonMusk", "Tesla Model S OTA update with Android Auto 5.2, fixes for memory leaks"));
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/MainActivity.java]: Android Auto 7.2

File: storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        tweets.add(Tweet.newTweet("Apple", "Yosemite update: fixes for Wifi issues, yosemite-wifi-patch#142"));
+
+        // Looks/reads nice, isn't it?
+        try {
+            PutResults<Tweet> results = storIOSQLite
+                .put()
+                .objects(tweets)
+                .prepare()
+                .executeAsBlocking();
+            Toast.makeText(this, getResources().getQuantityString(R.plurals.tweets_inserted, results.results().size()), Toast.LENGTH_LONG).show();
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/MainActivity.java]: Empty line would make it a little bit more readable

File: storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/TweetsAdapter.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.List;
+
+public class TweetsAdapter extends RecyclerView.Adapter<TweetsAdapter.ViewHolder> {
+
+    private List<Tweet> tweets;
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/TweetsAdapter.java]: `@Nullable`

File: storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/TweetsAdapter.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        notifyDataSetChanged();
+    }
+
+    @Override
+    public int getItemCount() {
+        return tweets == null ? 0 : tweets.size();
+    }
+
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/TweetsAdapter.java]: `@NonNull`

File: storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/TweetsAdapter.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        final Tweet tweet = tweets.get(position);
+
+        holder.authorTextView.setText("@" + tweet.author());
+        holder.contentTextView.setText(tweet.content());
+    }
+
+    static class ViewHolder extends RecyclerView.ViewHolder {
+
+        private TextView authorTextView;
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/TweetsAdapter.java]: `@NonNull final`

File: storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/TweetsAdapter.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+        private TextView contentTextView;
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/java/com/pushtorefresh/storio/basic_sample/TweetsAdapter.java]: `@NonNull final`

File: storio-basic-sample-app/src/main/res/layout/activity_main.xml [hunk 13]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/main/res/layout/activity_main.xml]: Why `RelativeLayout` required here? I think you can use `RecyclerView` directly

File: storio-basic-sample-app/src/test/java/com/pushtorefresh/storio/basic_sample/ExampleUnitTest.java [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+public class ExampleUnitTest {
```
Comment (artem-zinnatullin) [inline storio-basic-sample-app/src/test/java/com/pushtorefresh/storio/basic_sample/ExampleUnitTest.java]: Pls remove it, meaningless tests worse than no tests!

**Comment by skrzyneckik** [comment]
@artem-zinnatullin I applied all of changes which you suggested

**Comment by artem-zinnatullin** [comment]
Thanks a lot, @skrzyneckik!

**Pull request merged**